numba/targets/tupleobj.py

```diff
@@ -277,3 +277,19 @@ def tuple_to_tuple(context, builder, fromty, toty, val):
     items = [context.cast(builder, v, f, t)
              for v, f, t in zip(olditems, fromty, toty)]
     return context.make_tuple(builder, toty, items)
+
+
+#------------------------------------------------------------------------------
+# Methods
+
+@lower_builtin("tuple.index", types.BaseTuple, types.Any)
```

Comment:
I think that this could be written using the [high-level extension API `@overload_method`](http://numba.pydata.org/numba-doc/latest/extending/high-level.html#implementing-methods) which would reduce the amount of code.

Something like:
```python
@overload_method(types.BaseTuple, 'index')
def tuple_index(tup, value):
    def impl(tup, value):
        for i in range(len(tup)):
            if tup[i] == value:
                return i
        raise ValueError("tuple.index(x): x not in tuple")
    return impl
```
ought to work with no low level API calls needed.